apps/studio/src/server/modules/folder/__tests__/folder.router.test.ts

```diff
@@ -0,0 +1,634 @@
+      // Arrange
+      const invalidSiteId = 999
+      const { site, folder } = await setupFolder()
+      await setupAdminPermissions({
+        userId: session.userId,
+        siteId: site.id,
+      })
+      expect(site.id).not.toEqual(invalidSiteId)
+
+      // Act
+      const result = caller.editFolder({
+        siteId: String(invalidSiteId),
+        permalink: "test-folder",
+        title: "fake",
+        resourceId: folder.id,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "You do not have sufficient permissions to perform this action",
+        }),
+      )
+    })
+
+    it("should allow edits onto a folder regardless of the parent", async () => {
+      // Arrange
+      const { site, folder } = await setupFolder()
+      const { page } = await setupPageResource({
+        siteId: site.id,
+        resourceType: "Page",
+      })
+      await setupAdminPermissions({
+        userId: session.userId,
+        siteId: site.id,
+      })
+      await db.updateTable("Resource").set({ parentId: page.id }).execute()
+      const permalink = "tempora-link"
+
+      // Act
+      const result = await caller.editFolder({
+        siteId: String(site.id),
+        permalink,
+        resourceId: folder.id,
+        title: folder.title,
+      })
+
+      // Assert
+      const expected = await getFolderWithPermalink({
+        permalink,
+        siteId: site.id,
+      })
+      expect(result).toMatchObject({
+        id: expected.id,
+        title: expected.title,
+        permalink: expected.permalink,
+        parentId: page.id,
+      })
+    })
+
+    it("should throw 403 if user does not have access to the site", async () => {
+      // Arrange
+      const permalink = "test-folder-777"
+      const { site, folder } = await setupFolder()
+
+      // Act
+      const result = caller.editFolder({
+        siteId: String(site.id),
+        permalink,
+        resourceId: folder.id,
+        title: folder.title,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "You do not have sufficient permissions to perform this action",
+        }),
+      )
+    })
+
+    it("should allow edits on a root level folder regardless of the role", async () => {
+      // Arrange
+      const permalink = "test-folder-777"
+      const { site, folder } = await setupFolder()
+      await setupPermissions({
+        userId: session.userId,
+        siteId: site.id,
+        role: "Editor",
+      })
+
+      // Act
+      const result = await caller.editFolder({
+        siteId: String(site.id),
+        permalink,
+        resourceId: folder.id,
+        title: folder.title,
+      })
+
+      // Assert
+      expect(result).toMatchObject({ permalink, id: folder.id })
```

Comment:
to also check against result from `getFolderWithPermalink`?

Reply:
donezxc